osu.Game.Rulesets.Mania/Objects/Drawables/DrawableHoldNote.cs

```diff
@@ -323,7 +323,12 @@ public void OnReleased(KeyBindingReleaseEvent<ManiaAction> e)
 
             // do not run any of this logic when rewinding, as it inverts order of presses/releases.
             if (Time.Elapsed < 0)
+            {
+                // Except for the IsHitting state, as this handles animations that need to be reapplied
+                // after rewind.
+                isHitting.Value = false;
```

Comment:
I'd probably prefer this to live in `Update()`. There is precedent for this:

https://github.com/ppy/osu/blob/b5d9284de2d7054f9dc8200e6435cbc9420c9bc9/osu.Game.Rulesets.Mania/Objects/Drawables/DrawableHoldNote.cs#L219-L220

To me that'd be a lot safer and simpler - while I *think* your version works, it only works because of the inverted press/release order.

Reply:
Missed this review. Have applied a change in line with this which seems to still work quite well.